app/Database/CourseQueries.hs

```diff
@@ -271,3 +273,22 @@ queryGraphs =
         do graphs :: [Entity Graph] <- selectList [] []
            return $ createJSONResponse graphs
 
+-- ========================================================
+
+getLectureTime :: [String] -> IO (Maybe [Time])
+getLectureTime [code, section, session] = runSqlite databasePath $ do 
```

Comment:
Ah, I see what you're doing below. In this case, please have the type of the parameter be `(String, String, String)`.